### ISSUE TITLE
The initial loss of training on other dataset is very small

### ISSUE THREAD

**Issue by suyuzhang** [nianticlabs/monodepth2#369]
Hi,
Thanks for sharing awesome research.

I have some questions about training monodepth2 using the indoor dataset.
I used photos taken by the small car for training and set the FPS to 15.
The shooting environment is indoor.
My dataset looks like this format.
![image](https://user-images.githubusercontent.com/28804865/122520183-df250c00-d045-11eb-9e8b-c1e5714c8828.png)

I started training the monodepth2 from scratch, but the initial loss was very small.

`python train.py --model_name mono_640x480_fs_0618_k --frame_ids 0 -1 1 --height 480 --batch_size 6 --max_depth 10`
![image](https://user-images.githubusercontent.com/28804865/122514983-5a36f400-d03f-11eb-9333-df06c8e567f6.png)

I only change the data loader and command the check_depth function, because I don't have ground truth.

```

class CustomDataset(MonoDataset):
    """Superclass for different types of KITTI dataset loaders
    """
    def __init__(self, *args, **kwargs):
        super(KneronDataset, self).__init__(*args, **kwargs)

        # NOTE: Make sure your intrinsics matrix is *normalized* by the original image size.
        # To normalize you need to scale the first row by 1 / image_width and the second row
        # by 1 / image_height. Monodepth2 assumes a principal point to be exactly centered.
        # If your principal point is far from the center you might need to disable the horizontal
        # flip augmentation.
        self.K = np.array([[0.88, 0, 0.49, 0],
                           [0, 2.96, 1.31, 0],
                           [0, 0, 1, 0],
                           [0, 0, 0, 1]], dtype=np.float32)

        self.full_res_shape = (1242, 375)
        self.side_map = {"2": 2, "3": 3, "l": 2, "r": 3}

    def check_depth(self):
        # line = self.filenames[0].split()
        # scene_name = line[0]
        # frame_index = int(line[1])

        # velo_filename = os.path.join(
        #     self.data_path,
        #     scene_name,
        #     "velodyne_points/data/{:010d}.bin".format(int(frame_index)))

        # return os.path.isfile(velo_filename)
        return None
```

The following is my log:
[loss]
![image](https://user-images.githubusercontent.com/28804865/122517038-1a254080-d042-11eb-924c-72543c7ddc82.png)

[disparity map]
![image](https://user-images.githubusercontent.com/28804865/122517188-50fb5680-d042-11eb-97cc-c5b4b873ade3.png)

[predicted image]
![image](https://user-images.githubusercontent.com/28804865/122517336-74be9c80-d042-11eb-998d-ca4e46f89671.png)

[color image]
![image](https://user-images.githubusercontent.com/28804865/122518407-ba2f9980-d043-11eb-9691-05fdca0dd92d.png)
 
I analyzed it, but it was hard to find the cause of the small loss.
I have tried a larger frame step but still have the same problem.
Could you give me some advice?

**Comment by mdfirman** [comment]
Thanks for your interest here!

I would say that the data you are trying to train on is extremely challenging for self-supervised learning, as it has

* Large untextured regions
* Reflective surfaces

Overall, I would say that I never seen self-supervised learning work on this type of data. So unfortuntely I cannot give you any suggestions which I think would help.

The only thing I could suggest is to read through previous issues (e.g. this one https://github.com/nianticlabs/monodepth2/issues/343) to see if you can get any hints.

**Comment by suyuzhang** [comment]
Thanks for your response!

I increase the frame step and retrained a model.
It seems to get an acceptable result.
![244_disp](https://user-images.githubusercontent.com/28804865/122706074-aecdaf80-d289-11eb-80e7-2f2639248850.jpeg)

But the loss still has not converged.
![image](https://user-images.githubusercontent.com/28804865/122706326-49c68980-d28a-11eb-96c8-884eee851de9.png)

Have you seen similar behavior before?

**Comment by mdfirman** [comment]
Thanks for reporting back! I'm sorry to say that it doesn't look like your network is learning good depth at all. The floor for example should be showing a gradient in depth as it goes away from the camera.

In general then I think that your setup (and in particular your training data) isn't right for self-supervised learning

**Comment by VectNull** [comment]
> Hi, Thanks for sharing awesome research.
> 
> I have some questions about training monodepth2 using the indoor dataset. I used photos taken by the small car for training and set the FPS to 15. The shooting environment is indoor. My dataset looks like this format. ![image](https://user-images.githubusercontent.com/28804865/122520183-df250c00-d045-11eb-9e8b-c1e5714c8828.png)
> 
> I started training the monodepth2 from scratch, but the initial loss was very small.
> 
> `python train.py --model_name mono_640x480_fs_0618_k --frame_ids 0 -1 1 --height 480 --batch_size 6 --max_depth 10` ![image](https://user-images.githubusercontent.com/28804865/122514983-5a36f400-d03f-11eb-9333-df06c8e567f6.png)
> 
> I only change the data loader and command the check_depth function, because I don't have ground truth.
> 
> ```
> 
> class CustomDataset(MonoDataset):
>     """Superclass for different types of KITTI dataset loaders
>     """
>     def __init__(self, *args, **kwargs):
>         super(KneronDataset, self).__init__(*args, **kwargs)
> 
>         # NOTE: Make sure your intrinsics matrix is *normalized* by the original image size.
>         # To normalize you need to scale the first row by 1 / image_width and the second row
>         # by 1 / image_height. Monodepth2 assumes a principal point to be exactly centered.
>         # If your principal point is far from the center you might need to disable the horizontal
>         # flip augmentation.
>         self.K = np.array([[0.88, 0, 0.49, 0],
>                            [0, 2.96, 1.31, 0],
>                            [0, 0, 1, 0],
>                            [0, 0, 0, 1]], dtype=np.float32)
> 
>         self.full_res_shape = (1242, 375)
>         self.side_map = {"2": 2, "3": 3, "l": 2, "r": 3}
> 
>     def check_depth(self):
>         # line = self.filenames[0].split()
>         # scene_name = line[0]
>         # frame_index = int(line[1])
> 
>         # velo_filename = os.path.join(
>         #     self.data_path,
>         #     scene_name,
>         #     "velodyne_points/data/{:010d}.bin".format(int(frame_index)))
> 
>         # return os.path.isfile(velo_filename)
>         return None
> ```
> 
> The following is my log: [loss] ![image](https://user-images.githubusercontent.com/28804865/122517038-1a254080-d042-11eb-924c-72543c7ddc82.png)
> 
> [disparity map] ![image](https://user-images.githubusercontent.com/28804865/122517188-50fb5680-d042-11eb-97cc-c5b4b873ade3.png)
> 
> [predicted image] ![image](https://user-images.githubusercontent.com/28804865/122517336-74be9c80-d042-11eb-998d-ca4e46f89671.png)
> 
> [color image] ![image](https://user-images.githubusercontent.com/28804865/122518407-ba2f9980-d043-11eb-9691-05fdca0dd92d.png)
> 
> I analyzed it, but it was hard to find the cause of the small loss. I have tried a larger frame step but still have the same problem. Could you give me some advice?

Hello, I would like to ask you, how is the depth estimation in your environment solved? Thank you!

**Comment by VectNull** [comment]
Hello, I would like to ask you, how is the depth estimation in your environment solved? Thank you!

**Comment by hdacnw** [comment]
@wangsigong @suyuzhang May I know if you could kindly share potential solutions to the problem? Thanks!
